misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -399,6 +429,436 @@ public RecordFieldWithDefaultValueNode transform(RecordFieldWithDefaultValueNode
+
+        return assignmentStatementNode.modify()
+                .withVarRef(varRef)
+                .withEqualsToken(equalsToken)
+                .withExpression(expression)
+                .withSemicolonToken(semicolonToken)
+                .apply();
+    }
+
+    @Override
+    public CompoundAssignmentStatementNode transform(CompoundAssignmentStatementNode compoundAssignmentStatementNode) {
+        ExpressionNode lhsExpression = formatNode(compoundAssignmentStatementNode.lhsExpression(), 1, 0);
+        Token binaryOperator = formatToken(compoundAssignmentStatementNode.binaryOperator(), 0, 0);
+        Token equalsToken = formatToken(compoundAssignmentStatementNode.equalsToken(), 1, 0);
+        ExpressionNode rhsExpression = formatNode(compoundAssignmentStatementNode.rhsExpression(), 0, 0);
+        Token semicolonToken = formatToken(compoundAssignmentStatementNode.semicolonToken(),
+                this.trailingWS, this.trailingNL);
+
+        return compoundAssignmentStatementNode.modify()
+                .withLhsExpression(lhsExpression)
+                .withBinaryOperator(binaryOperator)
+                .withEqualsToken(equalsToken)
+                .withRhsExpression(rhsExpression)
+                .withSemicolonToken(semicolonToken)
+                .apply();
+    }
+
+    @Override
+    public DoStatementNode transform(DoStatementNode doStatementNode) {
+        boolean hasOnFailClause = doStatementNode.onFailClause().isPresent();
+
+        Token doKeyword = formatToken(doStatementNode.doKeyword(), 1, 0);
+        BlockStatementNode blockStatement = formatNode(doStatementNode.blockStatement(),
+                hasOnFailClause ? 1 : this.trailingWS, hasOnFailClause ? 0 : this.trailingNL);
+
+        if (hasOnFailClause) {
+            OnFailClauseNode onFailClause = formatNode(doStatementNode.onFailClause().get(),
+                    this.trailingWS, this.trailingNL);
+            doStatementNode = doStatementNode.modify().withOnFailClause(onFailClause).apply();
+        }
+
+        return doStatementNode.modify()
+                .withDoKeyword(doKeyword)
+                .withBlockStatement(blockStatement)
+                .apply();
+    }
+
+    @Override
+    public ForEachStatementNode transform(ForEachStatementNode forEachStatementNode) {
+        boolean hasOnFailClause = forEachStatementNode.onFailClause().isPresent();
+
+        Token forEachKeyword = formatToken(forEachStatementNode.forEachKeyword(), 1, 0);
+        TypedBindingPatternNode typedBindingPattern = formatNode(forEachStatementNode.typedBindingPattern(), 1, 0);
+        Token inKeyword = formatToken(forEachStatementNode.inKeyword(), 1, 0);
+        Node actionOrExpressionNode = formatNode(forEachStatementNode.actionOrExpressionNode(), 1, 0);
+        StatementNode blockStatement = formatNode(forEachStatementNode.blockStatement(),
+                hasOnFailClause ? 1 : this.trailingWS, hasOnFailClause ? 0 : this.trailingNL);
```

Comment:
moving this inside the below `if (hasOnFailClause) {` block would be cleaner..

Reply:
Done